checkout-core/src/main/java/com/adyen/checkout/core/model/JsonUtils.kt

```diff
@@ -0,0 +1,124 @@
+    return if (this.has(key)) this.getString(key) else null
+}
+
+fun JSONObject.getBooleanOrNull(key: String): Boolean? {
+    return if (this.has(key)) this.getBoolean(key) else null
+}
+
+fun JSONObject.toStringPretty(): String {
+    return try {
+        this.toString(INDENTATION_SPACES)
+    } catch (e: JSONException) {
+        PARSING_ERROR
+    }
+}
+
+fun JSONObject.optStringList(key: String): List<String>? {
+    return JsonUtils.parseOptStringList(this.optJSONArray(key))
+}
+
+class JsonUtils private constructor() {
+
+    init {
+        throw NoConstructorException()
+    }
+
+    companion object {
+        /**
+         * Writes a [JSONObject] to a [Parcel] as a [String].
+         *
+         * @param parcel The Parcel to be written to.
+         * @param jsonObject The JSONObject to be saved in the Parcel.
+         */
+        @JvmStatic
+        fun writeToParcel(parcel: Parcel, jsonObject: JSONObject?) {
+            if (jsonObject == null) {
+                parcel.writeInt(FLAG_NULL)
+            } else {
+                parcel.writeInt(FLAG_NON_NULL)
+                parcel.writeString(jsonObject.toString())
+            }
+        }
+
+        /**
+         * Reads a [JSONObject] previously saved on a [Parcel].
+         *
+         * @param parcel The Parcel to be read.
+         * @return The JSONObject that was contained in the Parcel.
+         */
+        @JvmStatic
+        @Throws(JSONException::class)
+        fun readFromParcel(parcel: Parcel): JSONObject? {
+            return when (parcel.readInt()) {
+                FLAG_NULL -> null
+                FLAG_NON_NULL -> JSONObject(parcel.readString() ?: "")
+                else -> throw IllegalArgumentException("Invalid flag.")
+            }
+        }
+
+        /**
+         * Parses a [JSONArray] to a list of Strings.
+         *
+         * @param jsonArray The JSONArray to be read.
+         * @return A [List] of strings, or null if the jsonArray was null.
+         */
+        @JvmStatic
+        fun parseOptStringList(jsonArray: JSONArray?): List<String>? {
+            if (jsonArray == null) {
+                return null
+            }
+            val list: MutableList<String> = ArrayList()
+            for (i in 0 until jsonArray.length()) {
+                val item = jsonArray.optString(i, null)
+                if (item != null) {
+                    list.add(item)
+                }
+            }
+            return Collections.unmodifiableList(list)
+        }
+
+        /**
+         * Serializes a List of String to a [JSONArray].
+         *
+         * @param stringList The [List] of Strings to be serialized.
+         * @return The populated [JSONArray]. Could be null.
+         */
+        @JvmStatic
+        fun serializeOptStringList(stringList: List<String?>?): JSONArray? {
+            if (stringList == null) {
+                return null
+            }
+            val jsonArray = JSONArray()
```

Comment:
you can do something like this:
```
return JSONArray().apply {
    stringList.filter { !it.isNullOrEmpty() }.forEach {
        put(it)
    }
}
```

Reply:
Oh yeah, I love how we can just replace all `TextUtils.isEmpty` with `isNullOrEmpty()` ❤️